pennylane/labs/resource_estimation/resource_utils.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility functions for resource estimation."""
+from collections.abc import Callable
+from typing import Any
+
+import numpy as np
+
+
+# pylint: disable=unnecessary-lambda-assignment, too-many-arguments, not-callable
+def approx_poly_degree(
+    x_vec: np.ndarray,
+    target_func: Callable,
+    error_tol: float = 1e-6,
+    degrees: tuple[int, int] | int | None = None,
+    basis: str | None = None,
+    loss_func: Callable | None = None,
+    fit_func: Callable | None = None,
+    project_func: Callable | None = None,
+    **fit_kwargs: dict[str, Any],
+):
+    r"""Compute the minimum degree of a polynomial that fits the data with a given error tolerance.
+
+    Args:
+        x_vec (np.ndarray): the domain values for sampling the target function ``target_func``.
+            The minimum length of ``x_vec`` is two and is expected to be sorted.
```

Comment:
Maybe out of scope for this PR, I'm not very familiar with the requirements. But do users ever want to use the domain [-infinity, infinity]?